### ISSUE TITLE
Standards - Long-Term Planning

### ISSUE THREAD

**Issue by ageecarroll** [PlantPhenoHack2017/HackTopics#3]
I'd like for those of us who will probably be doing more yacking and less hacking to consider how we might commonize our databases, metadata and ontologies. My proposed agenda is to grab as much low-hanging fruit as we can carry, and make action plans for addressing more complex topics down the line.

Consider strengths and gaps, and make plans for development if no existing solutions exist:
- Databases and architecture (BetyDB)
- Metadata (MIAPPE, ISA-TAB)
- Ontology systems 

(This is my first time using GitHub, hope I'm doing it right!)

**Comment by dlebauer** [comment]
You did it right! Nice work!

Once a common interchange format is identified, we can export data from BETYdb and other databases into that format. We've been looking at, BRAPI, ICASA, and more recently MIAPPE/ISA-TAB. At least one key industry partner has plans to support MIAPPE/ISA-TAB. For image / point cloud data we are developing formats that follow existing standards.

@craig-willis has been working on this on our end and has written up some useful information. It would be great to get feedback on this work. It could possibly be turned into a white paper or set of protocols that the community can build / adopt (Craig could you please link to your summaries).

**Comment by westerman** [comment]
People keep talking about MIAPPE/ISA-TAB as if they are one thing.  But from my reading MIAPPE is the standard for what data should be collected while ISA-TAB is the format for data exchange.  Are they that closely tied together that they should be referred as the same?   

My concern is if the up-stream database (say BetyDB) does not have all of the MIAPPE information in it then it becomes hard to create MIAPPE/ISA-TAB files.  Thus we should be talking about BetyDB supporting MIAPPE standards plus ISA-TAB formatting.  Or am I wrong in my understanding?  Would like to see what Craig has written.

**Comment by dlebauer** [comment]
If BETYdb doesn't meet standards we can update it or link with another database so that it does. But it is a good point that we can check.

**Comment by craig-willis** [comment]
I have limited design information in github related to our work with ICASA and the AgMIP JSON Data Objects: https://github.com/craig-willis/icasa/blob/master/docs/design.md.  We've had a number of meetings with folks from AgMIP/Crop Ontology/Agronomy Ontology groups.  We've been moving forward with generalizing the AgMIP JDO model for linked data.  I can imagine the resulting format confirming to MIAPPE, while also allowing us to incorporate information from other TERRA-related domains (e.g., sensors and computational models).

I am intrigued to see discussion of the ISA-TAB format, which didn't come much in earlier discussions.  I'm familiar with the ISA toolset and various MIA* standards, but have seen the JSON data objects approach as more fruitful for API-based systems like TERRA/Clowder/BETYdb. That said, it sounds like we need to revisit support for ISA-TAB if the community is moving in this direction.

**Comment by szarecor** [comment]
To chime in on what Rick said yesterday:

I'm new to the whole phenotyping world, but my understanding is as follows:
1. ISA defines a general and abstract data model. 
2. ISA-TAB is a file format based on the ISA data model.
3. In addition to ISA-TAB, there is an ISA-JSON spec as well. The two file formats should be completely interoperable.
3. The term ISA-TAB is often used imprecisely to refer to the data model and not the actual TSV file format.
4. MIAPPE is a data standard specific to plant phenotyping and the people behind it recommend ISA-TAB as the file format. In my mind, when they specify a specific ISA-* format, it doesn't matter much to me, as a programmer, if I can easily convert a tab to json and vice-versa. Personally, I'd rather consume/produce json files, but as long as the two are interoperable, it's a non-issue.

Does anyone want to chime in and correct any errors in the above assumptions?

In response to Craig's comment:

Maybe the hackathon could be used to explore ISA-JSON compatibility with the "API-based systems like TERRA/Clowder/BETYdb"?

**Comment by dlebauer** [comment]
@tinoDornbusch could you comment on LT work toward adopting ISA TAB?

**Comment by westerman** [comment]
Scott's 5 points reflect my understanding as well.   As for the use of 
ISA-TAB to refer to the data model, I believe that the ISA team is 
pulling away from that term in favor of just 'ISA'.  A post from the 
team on 10/28/16 said

"Since producing the linkedISA tool 
(https://github.com/ISA-tools/linkedISA), we had distinguished between 
the ISA abstract model and the different serialisations (Tab, RDF, and 
now also JSON)"

So in summary we have:

1) MIAPPE -- the minimum phenotyping  metadata needed to describe an 
experiment; data that conforms to MIAPPE could fit many data models

2) ISA - a way to model data into Investigation, Study, and Assays

3) ISA-TAB, ISA-JSON - file formats for import and export


An interesting web page on ISA-Tools is:

http://isa-tools.org/software-suite/

It shows various tools related to ISA.

**Comment by jshoyer** [comment]
The terminology is hard to keep straight, but I think you've got it.
1. [ISA Abstract Model](http://isa-specs.readthedocs.io/en/latest/isamodel.html)
2. [ISA-Tab serialization](http://isa-specs.readthedocs.io/en/latest/isatab.html), minimally three TSV files with names that by convention start `i_`, `s_`, and `a_`.
3. [ISA-JSON serialization](http://isa-specs.readthedocs.io/en/latest/isajson.html)
4. The abstract model and the ISA-Tab serialization have been conflated in the past.
    *  'ISA-Tab for phenotyping' is a particular way of using the ISA-Tab serialization. Configuration files (for the ISAcreator app) are available on the [downloads page](http://cropnet.pl/phenotypes/?page_id=71) of the the transPLANT site. [ISAcreator](https://github.com/ISA-tools/ISAcreator) is not being developed further; it seems that it has been superseded by the [ISA tools API](https://isatools.readthedocs.io/en/latest/) (isatools python package)
5. There is a MIAPPE checklist/table [PDF](http://cropnet.pl/phenotypes/wp-content/uploads/2016/04/MIAPPE.pdf) and further description by [Ćwiek-Kupczyńska et al.](http://doi.org/10.1186/s13007-016-0144-4) (2016). The ISA-Tab serialization has been used to implement one method for using the MIAPPE standard (via ISAcreator).

     

What is 'LT'?

**Comment by dlebauer** [comment]
Sorry ... LT is Lemnatec
